File: An 3/Sem 2/MPP/Java/Server/src/main/java/com/mcostea/SalesAgency/persistance/OrdersDAO.java
package com.mcostea.SalesAgency.persistance;

import com.mcostea.SalesAgency.model.Agent;
import com.mcostea.SalesAgency.model.Client;
import com.mcostea.SalesAgency.model.Order;
import com.mcostea.SalesAgency.model.Product;
import org.xml.sax.SAXException;
import org.xml.sax.helpers.DefaultHandler;

import javax.xml.parsers.SAXParser;
import javax.xml.parsers.SAXParserFactory;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.sql.Statement;
import java.util.ArrayList;
import java.util.jar.Attributes;
import java.util.logging.Level;
import java.util.logging.Logger;

public class OrdersDAO extends DAO {

    public ArrayList<Order> getOrders() {
        synchronized (this) {
            try {
                this.connect();
                String query = "SELECT orders.id, orders.quantity, orders.status, agents.id AS 'agentId', agents.name AS 'agentName', client.id AS 'clientId', client.name AS 'clientName', client.address AS 'clientAddress', products.id AS 'productId', products.name AS 'productName', products.price AS 'productPrice', products.quantity AS 'productQuantity'\n" +
                        "FROM orders\n" +
                        "JOIN agents ON agents.id=orders.agentId\n" +
                        "JOIN client ON client.id=orders.clientId\n" +
                        "JOIN products ON products.id=orders.productId";

                Statement stmt = this.conn.createStatement();
                ResultSet rs = stmt.executeQuery(query);

                ArrayList<Order> result = new ArrayList<Order>();

                while (rs.next()) {
                    Order order = new Order();
                    order.setOrderId(rs.getInt("id"));
                    order.setQuantity(rs.getInt("quantity"));
                    order.setStatus(rs.getString("status"));

                    Agent agent = new Agent();
                    agent.setAgentID(rs.getInt("agentId"));
                    agent.setName(rs.getString("agentName"));

                    Client client = new Client();
                    client.setClientID(rs.getInt("clientId"));
                    client.setName(rs.getString("clientName"));
                    client.setAddress(rs.getString("clientAddress"));

                    Product product = new Product();
                    product.setProductId(rs.getInt("productId"));
                    product.setName(rs.getString("productName"));
                    product.setQuantity(rs.getInt("productQuantity"));
                    product.setPrice(rs.getInt("productPrice"));

                    order.setAgent(agent);
                    order.setClient(client);
                    order.setProduct(product);

                    result.add(order);
                }

                rs.close();
                this.disconnect();
                return result;

            } catch (SQLException ex) {
                Logger.getLogger(OrdersDAO.class.getName()).log(Level.SEVERE, null, ex);
                return null;
            }
        }
    }

    public boolean updateOrder(Order o) {
        synchronized (this) {
            try {
                this.connect();
                String query = "UPDATE agency.orders SET productId=" + o.getProduct().getProductId() +
                        ", agentId=" + o.getAgent().getAgentID() + ", clientId=" + o.getClient().getClientID() +
                        ", quantity=" + o.getQuantity() + ", status=" + o.getStatus() +
                        "WHERE id=" + o.getOrderId();

                Statement stmt = this.conn.createStatement();
                int res = stmt.executeUpdate(query);

                System.out.println("Update result = " + res);

                this.disconnect();
                return true;

            } catch (SQLException ex) {
                Logger.getLogger(OrdersDAO.class.getName()).log(Level.SEVERE, null, ex);
                return false;
            }
        }
    }

    public boolean addOrder(Order o) {
        synchronized (this) {
            try {
                this.connect();
                String query = "INSERT INTO agency.orders VALUES (NULL, " + o.getProduct().getProductId() + ", " + o.getAgent().getAgentID() +
                        ", " + o.getClient().getClientID() + ", " + o.getQuantity() + ", '" + o.getStatus() + "')";

                Logger.getLogger(OrdersDAO.class.getName()).log(Level.INFO, query, (Object[]) null);

                Statement stmt = this.conn.createStatement();
                int res = stmt.executeUpdate(query);

                System.out.println("Insert result = " + res);

                this.disconnect();
                return true;

            } catch (SQLException ex) {
                Logger.getLogger(OrdersDAO.class.getName()).log(Level.SEVERE, null, ex);
                return false;
            }
        }
    }

    public boolean removeOrder(Order o) {
        synchronized (this) {
            try {
                this.connect();
                String query = "DELETE FROM agency.orders WHERE id=" + o.getOrderId();

                Statement stmt = this.conn.createStatement();
                int res = stmt.executeUpdate(query);

                System.out.println("Delete result = " + res);

                this.disconnect();
                return true;

            } catch (SQLException ex) {
                Logger.getLogger(OrdersDAO.class.getName()).log(Level.SEVERE, null, ex);
                return false;
            }
        }
    }

    public boolean getLoginResponse(Integer userId) {
        synchronized (this) {
            try {
                this.connect();
                String query = "Select * FROM Users WHERE ID ='" + userId + "'";

                Statement stmt = this.conn.createStatement();
                ResultSet rs = stmt.executeQuery(query);

                if (rs.next()) {
                    rs.close();
                    this.disconnect();
                    return true;
                }

                rs.close();
                this.disconnect();
                return true;

            } catch (SQLException ex) {
                Logger.getLogger(OrdersDAO.class.getName()).log(Level.SEVERE, null, ex);
                return false;
            }
        }
    }
}

File: An 3/Sem 2/MPP/CSharp/Server/Server/Persistance/OrdersDAO.cs
using System;
using System.Collections.Generic;
using System.Collections.Specialized;
using MySql.Data.MySqlClient;
using Model;

namespace Server
{
	public class OrdersDAO : DAO
	{
		public OrdersDAO ()
		{
		}

		public List<Order> GetOrders() {
			lock (this) {
				try {
					this.Connect();
					String query = "SELECT orders.id, orders.quantity, orders.status, agents.id AS 'agentId', agents.name AS 'agentName', client.id AS 'clientId', client.name AS 'clientName', client.address AS 'clientAddress', products.id AS 'productId', products.name AS 'productName', products.price AS 'productPrice', products.quantity AS 'productQuantity'\n" +
						"FROM orders\n" +
						"JOIN agents ON agents.id=orders.agentId\n" +
						"JOIN client ON client.id=orders.clientId\n" +
						"JOIN products ON products.id=orders.productId";

					MySqlCommand stmt = this.connection.CreateCommand ();
					stmt.CommandText = query;
					MySqlDataReader rs = stmt.ExecuteReader ();

					List<Order> result = new List<Order>();

					while (rs.Read()) {
						Order order = new Order();
						order.OrderId = rs.GetInt32 ("id");
						order.Quantity = rs.GetInt32("quantity");
						order.Status = rs.GetString("status");

						Agent agent = new Agent();
						agent.AgentID = rs.GetInt32("agentId");
						agent.Name = rs.GetString("agentName");

						Client client = new Client();
						client.ClientID = rs.GetInt32("clientId");
						client.Name = rs.GetString("clientName");
						client.Address = rs.GetString("clientAddress");

						Product product = new Product();
						product.ProductId = rs.GetInt32("productId");
						product.Name = rs.GetString("productName");
						product.Quantity = rs.GetInt32("productQuantity");
						product.Price = rs.GetInt32("productPrice");

						order.Agent = agent;
						order.Client = client;
						order.Product = product;

						result.Add (order);
					}

					rs.Close();
					this.Disconnect();
					return result;
				} catch (MySqlException ex) {
					Console.WriteLine (ex.ToString ());
					return null;
				}
			}
		}

		public Boolean UpdateOrder(Order o) {
			lock (this) {
				try {
					this.Connect();
					String query = "UPDATE agency.orders SET productId=" + o.Product.ProductId +
						", agentId=" + o.Agent.AgentID + ", clientId=" + o.Client.ClientID +
						", quantity=" + o.Quantity + ", status=" + o.Status +
						"WHERE id=" + o.OrderId;

					MySqlCommand stmt = this.connection.CreateCommand ();
					stmt.CommandText = query;

					int res = stmt.ExecuteNonQuery();

					Console.WriteLine("Update result = {0}", res);

					this.Disconnect();
					return true;

				} catch (MySqlException ex) {
					Console.WriteLine (ex.ToString ());
					return false;
				}
			}
		}

		public Boolean AddOrder(Order o) {
			lock (this) {
				try {
					this.Connect();
					String query = "INSERT INTO agency.orders VALUES (NULL, " + o.Product.ProductId + ", " + o.Agent.AgentID +
						", " + o.Client.ClientID + ", " + o.Quantity + ", '" + o.Status + "')";

					MySqlCommand stmt = this.connection.CreateCommand();
					stmt.CommandText = query;
					int res = stmt.ExecuteNonQuery();

					Console.WriteLine("Insert result = {0}", res);

					this.Disconnect();
					return true;

				} catch (MySqlException ex) {
					Console.WriteLine (ex.ToString ());
					return false;
				}
			}
		}

		public Boolean RemoveOrder(Order o) {
			lock (this) {
				try {
					this.Connect();
					String query = "DELETE FROM agency.orders WHERE id=" + o.OrderId;

					MySqlCommand stmt = this.connection.CreateCommand();
					stmt.CommandText = query;
					int res = stmt.ExecuteNonQuery();

					Console.WriteLine("Delete result = {0}", res);

					this.Disconnect();
					return true;

				} catch (MySqlException ex) {
					Console.WriteLine (ex.ToString ());
					return false;
				}
			}
		}
	}
}

